NwPluginAPI/Enums/ServerEventType.cs

```diff
@@ -607,39 +615,39 @@ public enum ServerEventType : int
 		/// Event executed when player tries to throw projectile like grenades.
 		/// </summary>
 		/// <remarks>
-		/// Parameters: <see cref="IPlayer"/> player, <see cref="ThrowableItem"/> item, <see cref="float"/> forceAmount, <see cref="float"/> upwardsFactor, <see cref="Vector3"/> torque, <see cref="Vector3"/> velocity.
+		/// Parameters: <see cref="IPlayer"/> player, <see cref="ThrowableItem"/> item, <see cref="ThrowableItem.ProjectileSettings"/> projectileSettings
 		/// </remarks>
 		PlayerThrowProjectile = 76,
 
 		/// <summary>
 		/// Event executed when player tries to activate SCP 914.
 		/// </summary>
 		/// <remarks>
-		/// Parameters: <see cref="IPlayer"/> player.
+		/// Parameters: <see cref="IPlayer"/> player, <see cref="Scp914KnobSetting"/> knobSetting.
 		/// </remarks>
 		Scp914Activate = 77,
 
 		/// <summary>
 		/// Event executed when player tries to change SCP 914 knob.
 		/// </summary>
 		/// <remarks>
-		/// Parameters: <see cref="IPlayer"/> player.
+		/// Parameters: <see cref="IPlayer"/> player, <see cref="Scp914KnobSetting"/> knobSetting, <see cref="Scp914KnobSetting"/> previousKnobSetting.
 		/// </remarks>
 		Scp914KnobChange = 78,
 
 		/// <summary>
 		/// Event executed when SCP 914 upgrades player inventory.
 		/// </summary>
 		/// <remarks>
-		/// Parameters: <see cref="IPlayer"/> player, <see cref="ItemBase"/> item.
+		/// Parameters: <see cref="IPlayer"/> player, <see cref="ItemBase"/> item, <see cref="ItemBase"/> newItem.
```

Comment:
Due to how item upgrading is handled by the game, adding that argument will require moving the entire event AFTER the item is upgraded and it will cause the event to be not cancellable. Maybe we should split it into two events (just like command events and coin flip event)?